src/main/java/games/strategy/util/Match.java

```diff
@@ -96,70 +89,45 @@ private Match(final Predicate<T> condition) {
 
   /**
    * returns true if all elements in the collection match.
+   * returns false if the collection is empty.
+   */
+  public static <T> boolean allMatchNotEmpty(final Collection<T> collection, final Match<T> match) {
```

Comment:
Yeah... not sure what to think about this.  When I first saw it, I thought it smelled because we are now leaking an implementation detail that might possibly be a bug.  However, after reviewing some of the call sites, there are definitely some cases where having an empty collection should fail the match.

Since we are modifying 80+ call sites with this change, I was going to suggest we extract the `&& !collection.isEmpty()` expression into the larger expression at the call site where this method is called.  That would put it square in the face of the next dev who has to review those Boolean conditions so that they have to think of the implications of an empty collection.  In a sense, I want to discourage calls to this method because the empty collection test seems like a concern that `Match` should not be worried about.

So, the more I think about it, the more I'm ok with adding this method.  I would suggest it possibly be renamed to `allMatchAndNotEmpty()`, the `And` making it a little easier to read and putting the dev on notice that this is a compound condition (similar to how Guava named `Strings#isNullOrEmpty()`).  Actually, `notEmptyAndAllMatch()` probably makes more sense, but it then doesn't really follow the method naming conventions in this class.  But maybe that's what we want to make a dev take notice?

@ron-murhammer Since you looked at `allMatch()` returning `false` for an empty collection recently, do you have any thoughts?